### ISSUE TITLE
Warning log when `history.replaceState()` is called frequently

### ISSUE THREAD

**Issue by chrisvariety** [brillout/vite-plugin-ssr#46]
Similar to https://github.com/nodejs/nodejs.dev/issues/917

Safari throws an error when scrolling up and down on any clientRouter enabled page.

```
[Error] SecurityError: Attempt to use history.replaceState() more than 100 times per 30 seconds
	replaceState (vite-plugin-ssr_client_router.js:1108)
	saveScrollPosition (vite-plugin-ssr_client_router.js:1108)
```

**Comment by brillout** [comment]
A throttle would do the trick. Is this production critical? So I can prioritize.

**Comment by brillout** [comment]
(replaceState() is called upon window.onscroll)

**Comment by chrisvariety** [comment]
not at all a priority, just logging.

**Comment by chrisvariety** [comment]
sorry didn't mean to scare you with the title, your revised version looks much better 😄

**Comment by brillout** [comment]
Thanks for caring 😊

> scare you 

I'm not easily scared ;-).

> your revised version looks much better 😄

😀 I changed it for others 🙂

**Comment by brillout** [comment]
Released in `0.1.0-beta.35`.

The fix uses a `100ms` throttle; there still can be more than 100 `history.replaceState()` calls within 30 seconds. (I'm not increasing the throttle above `100ms` for good reasons.)

The worst thing that can happen for Safari users is that `vite-plugin-ssr` restores a wrong scroll position, and I expect this to be very seldom.

Right now, `vite-plugin-ssr` has other higher prio stuff, so I'm inclined to not spend too much time for a perfect solution for the time being.

**Comment by brillout** [comment]
SvelteKit does scroll restoration without scroll listener; https://github.com/sveltejs/kit/pull/3938.
